uncompatible with react 17

while i was trying to install the module i get the uncompatible verssion with the newest react 17 version with the internal 16 version.
need a full update then or we go back to react 16 😩

I removed package-lock.json and run npm install -f but it still returning same error (ARE YOU SURE YOUR PROJECT IS UNDER REACT 17)
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: static-landing@0.1.0
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"17.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.12.0" from react-phone-input-2@2.13.9
npm ERR! node_modules/react-phone-input-2
npm ERR!   react-phone-input-2@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /.npm/_logs/2021-02-15T21_55_44_287Z-debug.log

Yes I am sure it is in react 17
An react-phone-input-2 is working fine for me


On Tue, Feb 16, 2021, 03:28 Adnane ROUHI <notifications@github.com> wrote:

> I removed package-lock.json and run npm install -f but it still returning
> same error (ARE YOU SURE YOUR PROJECT IS UNDER REACT 17)
> npm ERR! code ERESOLVE
> npm ERR! ERESOLVE unable to resolve dependency tree
> npm ERR!
> npm ERR! While resolving: static-landing@0.1.0
> npm ERR! Found: react@17.0.1
> npm ERR! node_modules/react
> npm ERR! react@"17.0.1" from the root project
> npm ERR!
> npm ERR! Could not resolve dependency:
> npm ERR! peer react@"^16.12.0" from react-phone-input-2@2.13.9
> npm ERR! node_modules/react-phone-input-2
> npm ERR! react-phone-input-2@"*" from the root project
> npm ERR!
> npm ERR! Fix the upstream dependency conflict, or retry
> npm ERR! this command with --force, or --legacy-peer-deps
> npm ERR! to accept an incorrect (and potentially broken) dependency
> resolution.
> npm ERR!
> npm ERR! See /.npm/eresolve-report.txt for a full report.
>
> npm ERR! A complete log of this run can be found in:
> npm ERR! /.npm/_logs/2021-02-15T21_55_44_287Z-debug.log
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/bl00mber/react-phone-input-2/issues/384#issuecomment-779466005>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIERYUUYFWYFGML4FPG3CWDS7GKIJANCNFSM4XRMUNZQ>
> .
>

--- a/package.json
+++ b/package.json
@@ -81,8 +81,8 @@
     "webpack-merge": "^4.2.1"
   },
   "peerDependencies": {
-    "react": "^16.12.0",
-    "react-dom": "^16.12.0"
+    "react": "^16.12.0 || ^17.0.0",
+    "react-dom": "^16.12.0 || ^17.0.0"
   },
   "dependencies": {
     "classnames": "^2.2.6",

